flink-runtime/src/main/java/org/apache/flink/runtime/highavailability/nonha/embedded/EmbeddedJobResultStore.java

```diff
@@ -20,112 +20,57 @@
 
 import org.apache.flink.annotation.VisibleForTesting;
 import org.apache.flink.api.common.JobID;
+import org.apache.flink.runtime.highavailability.AbstractThreadsafeJobResultStore;
 import org.apache.flink.runtime.highavailability.JobResultEntry;
 import org.apache.flink.runtime.highavailability.JobResultStore;
 import org.apache.flink.runtime.jobmaster.JobResult;
-import org.apache.flink.util.Preconditions;
-
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-import javax.annotation.concurrent.GuardedBy;
 
+import java.io.IOException;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.NoSuchElementException;
 import java.util.Set;
-import java.util.concurrent.locks.ReadWriteLock;
-import java.util.concurrent.locks.ReentrantReadWriteLock;
-import java.util.function.Supplier;
 import java.util.stream.Collectors;
 
 /** A thread-safe in-memory implementation of the {@link JobResultStore}. */
-public class EmbeddedJobResultStore implements JobResultStore {
-
-    private static final Logger LOG = LoggerFactory.getLogger(EmbeddedJobResultStore.class);
-
-    private final ReadWriteLock readWriteLock = new ReentrantReadWriteLock();
+public class EmbeddedJobResultStore extends AbstractThreadsafeJobResultStore {
 
-    @GuardedBy("readWriteLock")
-    @VisibleForTesting
-    final Map<JobID, JobResultEntry> dirtyJobResults = new HashMap<>();
+    @VisibleForTesting final Map<JobID, JobResultEntry> dirtyJobResults = new HashMap<>();
 
-    @GuardedBy("readWriteLock")
-    @VisibleForTesting
-    final Map<JobID, JobResultEntry> cleanJobResults = new HashMap<>();
+    @VisibleForTesting final Map<JobID, JobResultEntry> cleanJobResults = new HashMap<>();
```

Comment:
See my comment in `EmbeddedJobResultStoreInternal`. I guess, we can switch to `private` here without the `@VisibleForTesting` annotation.